cli/src/test/java/com/devonfw/tools/ide/common/SystemPathTest.java

```diff
@@ -33,4 +43,113 @@ public void systemPathShouldRecognizeNonWindowsPaths(String pathStringToTest) {
     assertThat(testResult).isFalse();
   }
 
+  @Test
+  public void testfindBinaryFindsBashBinaryOnWindowsUsingFilterExcludingWindowsAppsInPath() {
+    // arrange
+    String path = "project/workspaces";
+    IdeTestContext context = newContext("find-binary", path, false);
+    SystemInfo systemInfo = SystemInfoMock.of("windows");
+    context.setSystemInfo(systemInfo);
+    Predicate<Path> ignoreWindowsApps = p -> checkPathToIgnoreLowercase(p, "appdata\\local\\microsoft");
+    List<Path> paths = new ArrayList<>();
+    paths.add(context.getUserHome().resolve("AppData\\Local\\Microsoft\\WindowsApps"));
+    paths.add(context.getUserHome().resolve("PortableGit\\bin"));
+    SystemPath systemPath = new SystemPath(context, context.getSoftwarePath(), ';', paths);
+    Path bash = Path.of("bash");
+
+    // act
+    Path result = systemPath.findBinary(bash, ignoreWindowsApps);
+
+    // assert
+    assertThat(result).isEqualTo(context.getUserHome().resolve("PortableGit").resolve("bin").resolve("bash.exe"));
+  }
+
+  @Test
+  public void testfindBinaryFindsGraalvmBinaryOnWindowsUsingFilterExcludingJava() {
+    // arrange
+    String path = "project/workspaces";
+    IdeTestContext context = newContext("find-binary", path, false);
+    SystemInfo systemInfo = SystemInfoMock.of("windows");
+    context.setSystemInfo(systemInfo);
+    Predicate<Path> ignoreJava = p -> checkPathToIgnoreLowercase(p, "java\\bin");
+    SystemPath systemPath = new SystemPath(context, "", context.getIdeRoot(), context.getSoftwarePath(), ';', new ArrayList<>());
```

Comment:
I see 5 identical copy&paste code blocks like this:
How about extracting a private method:
```java
private static SystemPath newSystemPathFromSoftware() {
  IdeTestContext context = newContext("find-binary", "project/workspaces", false);
  SystemInfo systemInfo = SystemInfoMock.of("windows");
  context.setSystemInfo(systemInfo);
  String pathVariable = "";
  return new SystemPath(context, pathVariable, context.getIdeRoot(), context.getSoftwarePath(), ';', new ArrayList<>());
}
```

Reply:
I can't replace the occurences with your method because I still need to use the context for my assertions to determine the correct path.